src/core/server/legacy_compat/legacy_service.ts

```diff
@@ -121,22 +125,24 @@ export class LegacyService implements CoreService {
       // managed by ClusterManager or optimizer) then we won't have that info,
       // so we can't start "legacy" server either.
       serverOptions:
-        httpServerInfo !== undefined
+        deps.http !== undefined
           ? {
-              ...httpServerInfo.options,
-              listener: this.setupProxyListener(httpServerInfo.server),
+              ...deps.http.options,
+              listener: this.setupProxyListener(deps.http.server),
             }
           : { autoListen: false },
+      handledConfigPaths: await this.core.configService.getUsedPaths(),
```

Comment:
__note:__ I decided to keep this structure for now `(kbnServer.core --> { serverOptions, handledConfigPaths, plugins })` to group everything that core provides to the legacy world, but we can later split into `core` and `plugins` as we were discussing.